src/adapters/node.js

```diff
@@ -1,36 +1,53 @@
-// Generated by CoffeeScript 1.9.3
-(function() {
+(function () {
     var request = require('request');
     var mkFhir = require('../fhir');
     var Q = require('q');
 
     var adapter = {
         defer: Q.defer,
-        http: function(args) {
+        http: function (args) {
             var deff = Q.defer();
-            args.body = args.data;
+
+            if (args.resource != undefined) {
```

Comment:
My understanding is that either resource should be present or a body should be present but not both. Is that right? If so it might be a good idea to include an assertion here to this effect.

Reply:
Honestly I think the problem lies even deeper and I did not propose a correct solution.

In the original version (args.body = args.data) I get the following error:

>Failed to parse JSON content, error was: Content does not appear to be FHIR JSON, first non-whitespace character was: \'&quot;\' (must be \'{\')

when I log the debug message the objects looks like this
> DEBUG (request): POST http://localhost:8080/baseDstu3/Patient { resource: 
   { resourceType: 'Patient',
     text: 
      { status: 'generated',
        div: '<div xmlns="http://www.w3.org/1999/xhtml">Test Test</div>' },
     name: [ [Object] ],
     gender: 'male',
     birthDate: '1971-12-04',
     address: [ [Object] ] },
  debug: true,
  baseUrl: 'http://localhost:8080/baseDstu3',
  credentials: 'same-origin',
  defer: [Function: defer],
  http: [Function: http],
  data: '{"resourceType":"Patient","text":{"status":"generated","div":"<div xmlns=\\"http://www.w3.org/1999/xhtml\\">Test Test</div>"},"name":[{"family":"Test","given":"Test"}],"gender":"male","birthDate":"1971-12-04","address":[{"line":["Leidensweg 10"],"city":"Specimendorf","postalCode":"9876"}]}',
  headers: 
   { Accept: 'application/json',
     'Content-Type': 'application/json',
     Prefer: 'return=representation' },
  method: 'POST',
  url: 'http://localhost:8080/baseDstu3/Patient' }

The only way I could resolve it was directly referring the body to the resource, but this only valid in a post with a resource (but not in a post with an operation or with a patch operation).
So I guess I have to recheck why this error message with the quotation is coming.

Reply:
It looks like the args.data Element is somehow serialized as a string over the wire. When I add the following function everything is working:

args.body = JSON.parse(args.data);

might that be a possible solution? I'm a bit puzzled, because in native.js the body is directly assigned the args.data.